android/KMEA/app/src/main/java/com/tavultesoft/kmea/util/WebViewUtils.java

```diff
@@ -0,0 +1,104 @@
+  public static final String TAG = "WebViewUtils";
+
+  // Keyman Engine functionality based on Chrome version
+  public enum EngineWebViewVersionStatus {
+    DISABLED, // WebView doesn't support touch keyboard features
+    DEGRADED, // WebView supports touch keyboards but not LDML keyboards
+    FULL;     // WebView supports touch keyboards and LDML keyboards
+  }
+
+  private static final String CHROME_INSTALL_PATTERN_FORMATSTR = "^.*Chrome/([\\d\\.]+)\\s.*$";
+  private static final Pattern installPattern = Pattern.compile(CHROME_INSTALL_PATTERN_FORMATSTR);
+
+  /**
+   * Get the Keyman Engine mode based on the Chrome version.
+   * @param context       - The context
+   * @param chromeVersion - String of the device's Chrome version.
+   *                      If not provided, the device's Chrome version is queried
+   * @return EngineWebViewVersionStatus
+   */
+  public static EngineWebViewVersionStatus getEngineWebViewVersionStatus(Context context, String chromeVersion) {
+    if (context == null) {
+      return EngineWebViewVersionStatus.DISABLED;
+    }
+    if (chromeVersion == null || chromeVersion.isEmpty()) {
+      chromeVersion = getChromeVersion(context);
+    }
+
+    if (FileUtils.compareVersions(chromeVersion, "57.0") == FileUtils.VERSION_GREATER) {
+      return EngineWebViewVersionStatus.FULL;
+    } else if (FileUtils.compareVersions(chromeVersion, "37.0") == FileUtils.VERSION_GREATER) {
+      return EngineWebViewVersionStatus.DEGRADED;
+    }
+
+    return EngineWebViewVersionStatus.DISABLED;
+  }
+
+  // Inject a meta viewport tag into the head of the file if it doesn't exist
+  public static void injectViewport(WebView webView) {
+    if (webView != null) {
+      webView.loadUrl(
+        "javascript:(function() {" +
+          "if(document.head && !document.querySelectorAll('meta[name=viewport]').length) {"+
+          "let meta=document.createElement('meta');"+
+          "meta.name='viewport';"+
+          "meta.content='width=device-width, initial-scale=1';"+
+          "document.head.appendChild(meta);"+
+          "}"+
+          "})()"
+      );
+    }
+  }
+
+  // Blank the webView and destroy completely
+  // Reference: https://stackoverflow.com/questions/17418503/destroy-webview-in-android/17458577#17458577
+  public static void cleanup(WebView webView) {
+    if (webView != null) {
+      webView.loadUrl("about:blank");
+      ViewGroup viewGroup = (ViewGroup) webView.getParent();
+      if (viewGroup != null) {
+        viewGroup.removeView(webView);
+      }
+      webView.removeAllViews();
+      webView.destroy();
+      webView = null;
+    }
+  }
+
+  /**
+   * Get the Chrome version. Returns empty string if Chrome not installed.
+   * @param context - The context
+   * @return String
+   */
+  private static String getChromeVersion(Context context) {
+    if (context == null) {
+      KMLog.LogInfo(TAG, "Chrome not installed");
+      return "";
+    }
+
+    WebView mWebView = new WebView(context);
```

Comment:
Do you construct a new instance because we may not have an instance available? It just seems costly to do that, wonder if it is necessary :) (or do we just not worry?)

Reply:
I considered passing the Keyboard WebView as a parameter. But thinking ahead, we might not have that instance available during kmp installation.

Reply:
I guess we could reuse that instance if available or make a new one if not?